geopandas/tests/test_overlay.py

```diff
@@ -438,16 +456,17 @@ def test_overlay_strict(how, keep_geom_type, geom_types):
     points1 = GeoSeries([Point((2, 2)), Point((3, 3))])
     df4 = GeoDataFrame({"col4": [1, 2], "geometry": points1})
 
-    if geom_types == "polys":
-        result = overlay(df1, df2, how=how, keep_geom_type=keep_geom_type)
-    elif geom_types == "poly_line":
-        result = overlay(df1, df3, how=how, keep_geom_type=keep_geom_type)
-    elif geom_types == "poly_point":
-        result = overlay(df1, df4, how=how, keep_geom_type=keep_geom_type)
-    elif geom_types == "line_poly":
-        result = overlay(df3, df1, how=how, keep_geom_type=keep_geom_type)
-    elif geom_types == "point_poly":
-        result = overlay(df4, df1, how=how, keep_geom_type=keep_geom_type)
+    with warnings.catch_warnings(record=True):
```

Comment:
This might be overkill, or a sign that this warning is too noisy and thus shouldn't actually be implemented, but a handful of these were raising warnings as a result of this change.

Reply:
That is because the data we use are designed to produce slivers and geometry collections. That is okay.